### PR TITLE
[Testing] Explain how to add or remove data in a collection of forms

### PR DESCRIPTION
**Pull request by alexisLefebvre**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | no |
| Applies to | all (based on 2.3) |
| Fixed tickets | - |

I added a part that explain how to test a [_Collection of Forms_](http://symfony.com/doc/2.3/cookbook/form/form_collections.html).

### CODE REVIEW

**Comment by HeahDude** [comment]
Great, thanks!

**Comment by alexislefebvre** [comment]
Can someone please give an opinion on this PR?

File: book/testing.rst [hunk 1]
```diff
@@ -699,6 +699,48 @@ their type::
     PHP format (it converts the keys with square brackets notation - e.g.
     ``my_form[subject]`` - to PHP arrays).
 
+If you use a :doc:`Collection of Forms </cookbook/form/form_collections>`,
+you can't add fields to an existing form with
+``$form['task[tags][0][name]'] = 'foo';``, this results in an error
```
Comment (xabbuh) [inline book/testing.rst]: I think "this results in [...]" should start a new sentence.

**Comment by alexislefebvre** [comment]
@xabbuh Thanks, I followed your advice.

File: book/testing.rst [hunk 2]
```diff
@@ -699,6 +699,48 @@ their type::
...
+
+    // Submit the form with the existing and new values.
+    $crawler = $this->client->request($form->getMethod(), $form->getUri(), $values,
+        $form->getPhpFiles());
+
+    // The 2 tags have been added to the collection.
+    $this->assertEquals(2, $crawler->filter('ul.tags > li')->count());
+
+Where ``task[tags][0][name]`` is the name of a field created
+with Javascript.
```
Comment (xabbuh) [inline book/testing.rst]: JavaScript

**Comment by xabbuh** [comment]
@alexislefebvre I just left some really minor last comment (we can fix that during merging). Looks good so far. :+1:

**Comment by weaverryan** [comment]
This is great - it would be very difficult to figure out :). Thanks @alexislefebvre!

**Comment by alexislefebvre** [comment]
Thanks you all! :)

**Pull request merged**